src/engine/batch.cpp

```diff
@@ -49,6 +49,8 @@ void Batch::add(Sequence* sequence, uint32_t token_budget) {
   sequences_.push_back(sequence);
   token_budgets_.push_back(token_budget);
   budget_used_.push_back(0);
+
+  input_embedding_ = sequence->get_input_embedding();
```

Comment:
sounds one input embedding for each sequence, if so, it should be a vector of input embeddings for the batch.

Reply:
yes, in this pr only support one sequence + one image, and will support multiple sequences later.